hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -147,23 +184,21 @@ private void initIfNeeded() {
         }
       }
       timings.add(timer.endTimer());
-      LOG.info(String.format("Metadata read for key %s took [open, baseFileRead, logMerge] %s ms", key, timings));
+      LOG.info(String.format("Metadata read for key %s took [baseFileRead, logMerge] %s ms", key, timings));
       return Option.ofNullable(hoodieRecord);
     } catch (IOException ioe) {
       throw new HoodieIOException("Error merging records from metadata table for key :" + key, ioe);
-    } finally {
-      closeIfNeeded();
     }
   }
 
   /**
-   * Open readers to the base and log files.
+   * Returns the readers to the base and log files.
+   *
+   * If reuse is allowed then cached readers are returned. Otherwise new readers are opened.
    */
-  private synchronized void openFileSliceIfNeeded() throws IOException {
-    if (metadataConfig.enableReuse() && baseFileReader != null) {
-      // we will reuse what's open.
-      return;
-    }
+  private Pair<HoodieFileReader, HoodieMetadataMergedLogRecordScanner> openReaders() throws IOException {
+    HoodieFileReader baseFileReader = null;
```

Comment:
this is confusing to have same var names for both local variables and instance variables. Can you please change it. I was reviewing code assuming openReaders sets instance vars only to realize after a long time that these are local vars.

Reply:
may be we could name instance vars as cachedBaseFileReader and cachedLogFileReader. wdyt?

Reply:
There was a review comment earlier due to which I changed from cachedBaseFileReader. 

What about using localBaseFileReader and logRecordScanner?